src/OpenTelemetry.Instrumentation.GrpcNetClient/Implementation/GrpcClientDiagnosticListener.cs

```diff
@@ -180,6 +184,21 @@ public void OnStartActivity(Activity activity, object payload)
                 GrpcInstrumentationEventSource.Log.EnrichmentException(ex);
             }
         }
+
+        // See https://github.com/grpc/grpc-dotnet/blob/ff1a07b90c498f259e6d9f4a50cdad7c89ecd3c0/src/Grpc.Net.Client/Internal/GrpcCall.cs#L1180-L1183
+        // this makes sure that top-level properties on the payload object are always preserved.
+#if NET6_0_OR_GREATER
+        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "The event source guarantees that top level properties are preserved")]
+#endif
+        static bool TryFetchRequest(object payload, out HttpRequestMessage request)
+        {
+            if (!StartRequestFetcher.TryFetch(payload, out request) || request == null)
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
(nit) - applies to all these locations.

```suggestion
        static bool TryFetchRequest(object payload, out HttpRequestMessage request)
            => StartRequestFetcher.TryFetch(payload, out request) && request != null;
```